src/main.zig

```diff
@@ -148,6 +154,7 @@ const Command = struct {
 
         while (true) {
             replica.tick();
+            command.storage.tick();
```

Comment:
Let's move this within `replica.tick()` since replica ultimately is responsible for `storage`.

Reply:
:thinking: Right now, `storage.tick()` just calls `io.tick()`. There is 1 storage : 1 replica ratio, but if you (for some reason) had multiple replicas in a process, they would all share an IO, and you wouldn't want to tick IO that more than once per replica tick.

(Not that you would ever actually want to run multiple replicas in a single process (outside of the simulator), I'm just trying to convey the abstractions. This is mostly in the interest of making TigerBeetle library-friendly)

Maybe replica should call `storage.tick()`, but `io.tick()` should move out of storage, to the top-level loop? It is a little weird that it is in `Storage` when the `MessageBus` also uses `IO`.

Reply:
> Maybe replica should call storage.tick(), but io.tick() should move out of storage, to the top-level loop? It is a little weird that it is in Storage when the MessageBus also uses IO.

Exactly—I think your suggestion here is the right way to solve it.